### PR TITLE
Fix restic on i386

### PR DESCRIPTION
**Pull request by fd0**
This fixes restic for Linux/i386 and instructs travis to run the tests also on i386.

### CODE REVIEW

**Comment by fw42** [comment]
I don't understand what the problem is and I don't understand why this PR fixes it. Can you explain?

**Comment by fw42** [comment]
is this about the id being uint32 and not uint64 or something like that?

File: snapshot.go [hunk 1]
```diff
@@ -71,10 +71,10 @@ func (sn Snapshot) ID() backend.ID {
 func (sn *Snapshot) fillUserInfo() error {
 	usr, err := user.Current()
 	if err != nil {
-		return err
+		return nil
```
Comment (fw42) [inline snapshot.go]: I don't understand this change

**Comment by fd0** [comment]
Oh, I'm sorry: I discovered that restic stopped working on Linux/i386, because several functions from the stdlib weren't implemented on that platform (e.g `user.Current()` and `user.LookupId()`). So the purpose of this PR is twofold: Add tests on i386 to travis, and make most cases with these lookup-functions not fail hard.

File: snapshot.go (hunk 1 above)
Comment (fd0) [inline snapshot.go]: If the current user cannot be determined, do not fill in any user info. This information isn't  required for restic to function (it's just an information after all, nothing that you'd really need to restore a backup), so don't fail hard when `user.Current()` returns an error.

**Comment by fd0** [comment]
Ah, I asked about these not-implemented functions (because it bothered me a lot), turns out, this functionality needs CGO (because it uses the libc functions), and that's disabled when cross-compiling: https://github.com/golang/go/issues/10868

**Comment by fw42** [comment]
Ok I get it now. Weird. But ok. :rocket:

**Pull request merged**